datafusion/sql/src/parser.rs

```diff
@@ -519,12 +519,7 @@ impl<'a> DFParser<'a> {
             Token::SingleQuotedString(s) => Ok(Value::SingleQuotedString(s)),
             Token::DoubleQuotedString(s) => Ok(Value::DoubleQuotedString(s)),
             Token::EscapedStringLiteral(s) => Ok(Value::EscapedStringLiteral(s)),
-            Token::Number(ref n, l) => match n.parse() {
```

Comment:
`n` is already a string, so calling `parse` on it to turn into a `String` always succeeds (and `e` is Unfallible). We can just use `n` directly here and save the ceremony